blog/2022-07-18-building-addon-introduction.md

```diff
@@ -0,0 +1,301 @@
+---
+title: "An Introduction to KubeVela Addons: Build Your Own Addon"
+authors:
+- name: Charlie Chiang
+  title: KubeVela Team
+  url: https://github.com/charlie0129
+  image_url: https://github.com/charlie0129.png
+tags: [ "KubeVela", "addon", "extensibility" ]
+description: Introduction to Building Addons
+image: https://raw.githubusercontent.com/oam-dev/KubeVela.io/main/docs/resources/KubeVela-03.png
+hide_table_of_contents: false
+---
+
+As we all know, KubeVela is a highly extensible platform, on which users can build their own customizations using [Definitions](https://kubevela.io/docs/platform-engineers/oam/x-definition). KubeVela addons are a convenient way to package all these customizations and their dependencies together, to extend the capabilities of KubeVela.
+
+This blog introduces the main concepts of an addon and guides you to quickly start building an addon.
+
+<!--truncate-->
+
+## Why use addons
+
+We typically use addons with [addon catalog](https://github.com/kubevela/catalog), which contains addons with all kinds of customizations that the KubeVela community carefully crafted. You can download and install these addons with just one click. For example, you can provide the ability to run Helm Components in your KubeVela Application to your cluster by installing FluxCD addon.
+
+Contrary to the convenience of one-key installation, without KubeVela addons, you have to install FluxCD in this way:
+
+> Actually, this is how we install FluxCD before KubeVela v1.1
+
+1. Install FluxCD using Helm Charts or downloaded yaml manifests
+2. Manually download FluxCD-related Definition manifests and apply them
+
+Although it seems like only 2 steps is needed, we found it to be quite troublesome:
+
+1. Complicated installation: users are required to refer to documentations to manually install FluxCD and tackle possible errors
+2. Scattered resources: users need to obtain different resource files from different places
+3. Hard distribution: users having to download manifests manually makes it hard to distribute all these resources in a uniformed way
+4. No multi-cluster support: KubeVela emphasizes multi-cluster delivery a lot. However, manual installation makes multi-cluster support hard to carry out.
```

Comment:
5. Version management: users need to manage the relationship between definitions, controllers and corresponding versions by themselves.